### PR TITLE
Add cmake for logging sample using a release

### PR DESCRIPTION
**Pull request by achamayou**
This shows how to build a C++ app out of tree against a release of CCF. It's tested by `./test_install_build.sh`, to make sure the release we have produced is usable/not missing headers etc.

I'll have a look at updating the documentation.

### CODE REVIEW

File: samples/apps/logging/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+# TODO: Should be obtained through FindPackage
+include(${CCF_DIR}/cmake/ccf.cmake)
+include(${CCF_DIR}/cmake/ccf_app.cmake)
+include(${CCF_DIR}/cmake/tools.cmake)
+
+include_directories(
+  ${CCF_DIR}/include/ccf ${CCF_DIR}/include/3rdparty ${CCF_DIR}/generated
+  ${CCF_DIR}/include/3rdparty/msgpack-c
+  ${CCF_DIR}/include/3rdparty/flatbuffers/include
```
Comment (achamayou) [inline samples/apps/logging/CMakeLists.txt:28]: NOPE

File: CMakeLists.txt [hunk 2]
```diff
@@ -2,6 +2,8 @@
 # Licensed under the Apache 2.0 License.
 cmake_minimum_required(VERSION 3.11)
 
+include(GNUInstallDirs)
```
Comment (achamayou) [inline CMakeLists.txt:5]: Necessary for a bunch of directory variable expansion in installed files.

File: CMakeLists.txt [hunk 3]
```diff
@@ -166,7 +168,11 @@ if("virtual" IN_LIST COMPILE_TARGETS)
   add_dependencies(ccf ccf.virtual)
 endif()
 
-install(EXPORT ccf DESTINATION cmake)
+install(
+  EXPORT ccf
+  DESTINATION cmake
+  FILE ccf-targets.cmake
```
Comment (achamayou) [inline CMakeLists.txt:174]: Exporting that as "ccf.cmake" caused a double include, and loads of unpleasant errors.

File: cmake/ccf-config.cmake.in [hunk 4]
```diff
@@ -0,0 +1,15 @@
+@PACKAGE_INIT@
+
+set(CCF_DIR "@CMAKE_INSTALL_PREFIX@")
+
+include("${CCF_DIR}/cmake/ccf-targets.cmake")
+include("${CCF_DIR}/cmake/preproject.cmake")
+include("${CCF_DIR}/cmake/ccf_app.cmake")
+include("${CCF_DIR}/cmake/tools.cmake")
+
+include_directories(
```
Comment (achamayou) [inline cmake/ccf-config.cmake.in:10]: This is not good, we want to get rid of that, looking at it.

**Comment by achamayou** [comment]
This works pretty well now, provided the user export CC/CXX to clang-8, ie. the stuff we otherwise do in preproject.cmake. I don't think we're going to be able to work around that very much, other than disabling LVI mitigations when we detect that's not the case. I'm open to ideas though.

File: CMakeLists.txt [hunk 5]
```diff
@@ -884,3 +894,23 @@ if(BUILD_TESTS)
...
+configure_package_config_file(
+  ${CMAKE_CURRENT_SOURCE_DIR}/cmake/ccf-config.cmake.in
+  ${CMAKE_BINARY_DIR}/cmake/ccf-config.cmake
+  INSTALL_DESTINATION
+  ${CMAKE_INSTALL_PREFIX}/cmake
+  PATH_VARS
+  CMAKE_INSTALL_LIBDIR
+  CMAKE_INSTALL_BINDIR
+  CMAKE_INSTALL_DATADIR
+  CMAKE_INSTALL_INCLUDEDIR)
```
Comment (jumaffre) [inline CMakeLists.txt:909]: Do we need to pass all of these here?

Comment (achamayou) [reply CMakeLists.txt:909]: Not DATADIR, which I'm removing, but the others I believe we do, otherwise they won't expand correctly.

**Review by jumaffre — APPROVED** [review]

**Pull request merged**